core/arch/arm/plat-vexpress/main.c

```diff
@@ -194,15 +194,44 @@ service_init(init_tzc400);
 #endif /*CFG_TZC400*/
 
 #if defined(PLATFORM_FLAVOR_qemu_virt)
-int psci_cpu_on(uint32_t core_id, uint32_t entry, uint32_t context_id)
+static void release_secondary_early_hpen(size_t pos)
 {
-	size_t pos = get_core_pos_mpidr(core_id);
-	uint32_t *sec_entry_addrs = phys_to_virt(SECRAM_BASE, MEM_AREA_IO_SEC);
-	static bool core_is_released[CFG_TEE_CORE_NB_CORE];
+	uint32_t *sec_entry_addrs;
+#ifdef CFG_VEXPRESS_BOOT_FROM_TFA
```

Comment:
What's `TFA`

Reply:
Sorry, should be `_ATF` or `_ARM_TF`. `CFG_WITH_ARM_TRUSTED_FW` is already defined and specifically denotes "Support from Arm trusted firmware BL31 services". I don't want to reuse it nor change its name.

Or maybe I should introduce `CFG_VEXPRESS_BOOT_FROM_BIOS_QEMU_TZ_ARM` ?
Or maybe just simply (propose to) discard old  **bios_qemu_tz_arm** mailbox support. To enable such support in our optee deliveries, one currently has to hack into our **build/qemu.mk** script to manually add the `-smp [0|1|2|3]` argument to `qemu-system-arm`.

Reply:
+1 to discard old bios_qemu_tz_arm mailbox support, since we don't have any magic that makes OP-TEE to just work with either old bios_qemu_tz_arm or new ARM-TF.

Reply:
ok. thanks.